### PR TITLE
chore(deletions) Move ScheduledDeletion models to deletions

### PR DESCRIPTION
**Pull request by markstory**
Move the `ScheduledDeletion` model classes into `sentry.deletions` so that all the deletion code is co-located.

Refs #77479

### CODE REVIEW

File: src/sentry/deletions/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from sentry.deletions.models.scheduleddeletion import (
+    RegionScheduledDeletion,
+    ScheduledDeletion,
+    get_regional_scheduled_deletion,
```
Comment (Christinarlong) [inline src/sentry/deletions/models/__init__.py:4]: nit: I think we only need to shim `RegionScheduledDeletion` for getsentry

**Review by Christinarlong — APPROVED** [review]
Do we need to add `sentry.deletions` to `server.py`/`INSTALLED_APPS`?

**Review by wedamija — COMMENTED** [review]
Is the longer term plan to move the migrations for these into a separate app as well?

**Comment by markstory** [comment]
> Is the longer term plan to move the migrations for these into a separate app as well?

I don't think it makes sense to move existing migrations, but future migrations can certainly be split out.

My near term goals are to create more application domains that use django-apps. While some teams are building new functionality with app style organization, there is a ton of historical code. We'll need to incrementally chip away at that existing code to reorganize and reduce the scope of `sentry.models` and `sentry.api`. Deletions, hybridcloud and control silo services all have clear boundaries around them which make them easy targets for refactoring :smile:

**Pull request merged**